### PR TITLE
Improve regression

### PR DESCRIPTION
**Pull request by kunjsong01**
Based on the comments, split PR #625 into multiple small PRs. 

This PR contains the changes for regression scripts introducing some new features: 
1. Test log retention
2. Provide additional command line args for each test case generated

Usage: 

Use `--old-frontend` to run regression with old frontend:
```
python3 testing_tool.py \
--regression esbmc-cpp/cpp \
--tool ../../release/bin/esbmc --mode ALL \
--old-frontend
```

Add `--retain-logs` to to keep the logs for all test cases to obtain the passing rate and error signatures.

### CODE REVIEW

**Review by lucasccordeiro — APPROVED** [review]
LGTM.

File: src/util/message/message.h [hunk 1]
```diff
@@ -30,18 +30,18 @@ class messaget
...
-   * 
+   *
    * @param message string to be printed
    * @param location location where the message happened
    */
   virtual void
   error(const std::string &message, const std::string location = "") const
   {
-    print(VerbosityLevel::Error, message, location);
+    print(VerbosityLevel::Error, "###ERROR: " + message, location);
```
Comment (mikhailramalho) [inline src/util/message/message.h]: Can you add a space between the `###` and the `ERROR`?

**Review by mikhailramalho — APPROVED** [review]
small nit, otherwise LGTM

**Review by rafaelsamenezes — APPROVED** [review]
LGTM

**Comment by fbrausse** [comment]
Note that `--old-frontend` (as well as other flags for `esbmc`) could as well be passed via the `ESBMC_OPTS` environment variable introduced in 9e5b8fbe95.

**Pull request merged**